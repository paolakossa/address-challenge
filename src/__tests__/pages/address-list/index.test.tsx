import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { useRouter } from "next/router";
import AddressList from "@/pages/address-list";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("AddressList", () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it("should render correctly", () => {
    render(<AddressList />);
    expect(screen.getByText("Lista de Endereços")).toBeInTheDocument();
    expect(screen.getByText("Nenhum endereço cadastrado.")).toBeInTheDocument();
  });

  it("should display addresses stored in localStorage", async () => {
    const mockAddresses = [
      {
        cep: "74070-010",
        logradouro: "Rua Teste",
        bairro: "Bairro Teste",
        localidade: "Cidade Teste",
        uf: "TT",
      },
    ];
    localStorage.setItem("addresses", JSON.stringify(mockAddresses));

    render(<AddressList />);

    await waitFor(() => {
      expect(screen.getByText("74070-010")).toBeInTheDocument();
      expect(screen.getByText("Rua Teste")).toBeInTheDocument();
      expect(screen.getByText("Bairro Teste")).toBeInTheDocument();
      expect(screen.getByText("Cidade Teste")).toBeInTheDocument();
      expect(screen.getByText("TT")).toBeInTheDocument();
    });
  });

  it("should navigate to the form page when clicking the button", () => {
    const pushMock = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({ push: pushMock });

    render(<AddressList />);

    fireEvent.click(screen.getByText("Voltar ao formulário"));

    expect(pushMock).toHaveBeenCalledWith("/form");
  });

  it("must match the snapshot", () => {
    const { asFragment } = render(<AddressList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
