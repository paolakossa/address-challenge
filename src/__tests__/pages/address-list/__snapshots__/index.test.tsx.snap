// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressList must match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-screen flex flex-col items-center bg-[#f4fdf8] p-6"
  >
    <button
      class="mb-6 px-6 py-2 bg-[#225741] text-white font-semibold rounded-lg hover:bg-green-700 transition"
      type="button"
    >
      <div
        class="flex items-center"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5 8.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="mr-2"
        >
          Voltar ao formulário
        </span>
      </div>
    </button>
    <div
      class="bg-white shadow-lg rounded-lg p-6 w-full max-w-5xl"
    >
      <h1
        class="text-2xl font-bold mb-4 text-center text-[#225741]"
      >
        Lista de Endereços
      </h1>
      <p
        class="text-center text-gray-500"
      >
        Nenhum endereço cadastrado.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
